sgx-hasher/src/lib.rs

```diff
@@ -77,6 +81,68 @@ impl Hasher {
     }
 }
 
+pub fn to_array(bn: bn::BigNum) -> [u8; 384] {
+    let mut array = [0u8; 384];
+    array.copy_from_slice(&bn.to_vec());
+    array
+}
+
+pub fn sign(hash: &[u8; 32], key: &rsa::Rsa<Private>) -> Result<sig::Signature> {
+    let author = sig::Author::new(sig::Vendor::default(), 0u32, 0u32);
+
+    let contents = sig::Contents::new(
+        misc::MiscSelect::EXINFO,
+        attr::Attributes::new(attr::Flags::MODE_64_BIT, attr::Xfrm::all()),
+        *hash,
+        0,
+        0,
+    );
+
+    let rsa_key = PKey::from_rsa(key.to_owned())?;
```

Comment:
Instead of cloning the key here, change the function signature. The caller can decide whether to clone or move.